pkgs/stdenv/generic/default.nix

```diff
@@ -147,7 +161,15 @@ let
           nativeBuildInputs = nativeBuildInputs ++ (if crossConfig == null then buildInputs else []);
           propagatedNativeBuildInputs = propagatedNativeBuildInputs ++
             (if crossConfig == null then propagatedBuildInputs else []);
-        })) (
+
+          __impureHostDeps = lib.unique (lib.sort (x: y: x < y) (computedImpureHostDeps ++ computedPropagatedImpureHostDeps ++ __propagatedImpureHostDeps ++ __impureHostDeps ++ __extraImpureHostDeps ++ [
```

Comment:
Note that `lib.unique` is an extremely slow function (O(n^2) iirc) so it definitely should not be used on a critical path (i.e. anywhere in `stdenv.mkDerivation`). `lib.sort` is probably not very fast either.